examples/resp-headings/index.js

```diff
@@ -0,0 +1,37 @@
+const plugin = require('tailwindcss/plugin')
+
+module.exports = plugin(({ addComponents, e, theme }) => {
+  const screens = theme('screens')
+  const pluginConfig = theme('headings', {})
+
+  const component = Object.entries(pluginConfig).map(([name, props]) => {
+    const { sizes, letterSpacing = '0', lineHeight, weight } = props
+    const mediaQueries = []
+
+    for (const key in screens) {
+      if (key in sizes) {
+        mediaQueries.push({
+          [`@media (min-width: ${screens[key]})`]: {
+            [`.heading-${name}`]: {
```

Comment:
Maaaybe escape the class names just in case? Although this plugin would be very unlikely to be supplied keys with special characters in them.